lib/auth/password.go

```diff
@@ -171,9 +171,9 @@ func (a *Server) ChangePassword(ctx context.Context, req *proto.ChangePasswordRe
 func (a *Server) checkPasswordWOToken(user string, password []byte) error {
 	const errMsg = "invalid username or password"
 
-	err := services.VerifyPassword(password)
-	if err != nil {
-		return trace.BadParameter(errMsg)
+	if len(password) > defaults.MaxPasswordLength {
```

Comment:
I would ditch this altogether or move this logic to a named function, parallel to VerifyPassword. As it stands now it's not clear why VerifyPassword isn't used and someone might be tempted to add it back.

+1 for regression testing too - we did cause a bug, so capturing it in a test seems wise. (As in: write a "small" password to storage and make sure it can be used to authenticate.)